Adds ListCapabilities on the D-BUS API

tobsan opened the pull request:
Both unit tests, service tests and user docs are updated as part of
this.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

Commit:
Adds ListCapabilities on the D-BUS API

Both unit tests, service tests and user docs are updated as part of
this.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>

thenor commented on agent/src/softwarecontaineragentadaptor.cpp:
```diff
@@ -73,6 +73,13 @@ bool SoftwareContainerAgentAdaptor::SetGatewayConfigs(
+                                                     bool &success)
+{
+    capabilities = m_agent.listCapabilities();
+    success = true; // This operation can not fail
```
Why do you want/need a success flag if it can't fail?

tobsan replied:
Mostly for consistency in the API. Everywhere where we have some value returned, we also return whether or not the call was successful. This is to avoid relying on magic values.

DunderRoffe replied:
That does not seem like a good reason to me.
I think there should always be a purpose to a return value otherwise users might end up getting false positives.

tobsan replied:
I've updated this PR where the success is removed both for ListCapabilities and for List (listing of containers). Neither change the state of anything, so this should be safe.

breakreturn commented on agent/unit-test/configstore_unittest.cpp:
```diff
@@ -118,6 +118,20 @@ TEST_F(ConfigStoreTest, readConfigFetchOneCap) {
+ * result in getting all those names that are listed in the manifest file.
+ */
+TEST_F(ConfigStoreTest, readConfigFetchAllCaps) {
+    FilteredConfigStore cs = FilteredConfigStore(testDataDir + sm_path);
```
Please take the opportunity to change the variables that are named differently than the others (and also not according to the style guide).

Commit:
ListCapabilities and List don't return success

Since these two methods doesn't change the state of the container,
and since they can't fail, it makes little sense to have a success
field that would always be set to true.

Signed-off-by: Tobias Olausson <tobias.olausson@pelagicore.com>
